src/wallet/wallet.cpp

```diff
@@ -4037,33 +4017,32 @@ const CAddressBookData* CWallet::FindAddressBookEntry(const CTxDestination& dest
 bool CWallet::UpgradeWallet(int version, std::string& error, std::vector<std::string>& warnings)
 {
     int prev_version = GetVersion();
-    int nMaxVersion = version;
-    if (nMaxVersion == 0) // the -upgradewallet without argument case
+    if (version == 0) // the -upgradewallet without argument case
```

Comment:
nit in commit d0982de633783ba285772d159e87672762e0c400

`-upgradewallet` has been removed

```suggestion
    if (version == 0) {
```

Or, if you feel like it you may also just cherry-pick the commits from #18730

Reply:
I've cherry picked those commits.